samcli/cli/cli_config_file.py

```diff
@@ -123,6 +124,51 @@ def __call__(self, config_path: Path, config_env: str, cmd_names: List[str]) ->
         return resolved_config
 
 
+def handle_parse_options(resolved_config: dict) -> None:
+    """
+    Click does some handling of options to convert them to the intended types.
+    When injecting the options to click through a samconfig, we should do a similar
+    parsing of the options to ensure we pass the intended type.
+
+    E.g. if multiple is defined in the click option but only a single value is passed,
+    handle it the same way click does by converting it to a list first.
+
+    Mutates the resolved_config object
+
+    Parameters
+    ----------
+    resolved_config: dict
+        Configuration options extracted from the configuration file
+    """
+    options_map = get_options_map()
+    for config_name, config_value in resolved_config.items():
+        if config_name in options_map:
+            try:
+                allow_multiple = options_map[config_name].multiple
+                if allow_multiple and not isinstance(config_value, list):
+                    resolved_config[config_name] = [config_value]
```

Comment:
Can we add a debug log here so that we will know that we altered the original value?

Reply:
Done